messages/src/lib.rs

```diff
@@ -86,7 +70,9 @@ impl Decode for Url {
 
 impl Display for Url {
     fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
-        (&self.0 as &dyn Display).fmt(f)
+        // This isn't actually lossy, since this type cannot be constructed from
+        // non-ASCII bytes.
+        write!(f, "{}", String::from_utf8_lossy(&self.0))
```

Comment:
Since this method returns `std::fmt::Result` anyway, could we use `String::from_utf8` and map errors `std::fmt::Error`?

Reply:
[`std::fmt::Result`](https://doc.rust-lang.org/std/fmt/type.Result.html) is just `pub type Result = Result<(), Error>;` and `std::fmt::Error` in turn is just `pub struct Error` so I think `.map_err(|_| std::fmt::Error)` should work. If not, then this is fine.

Reply:
Aha, that works. I was trying to make use of the `|_|`.